Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInputField.swift

```diff
@@ -197,15 +205,49 @@ struct CreditCardInputField: View {
         }
     }
 
+    func handleDidEndEditing(_ textToCheck: String? = nil) {
+        if textToCheck != nil {
+            self.text = textToCheck!
```

Comment:
@petruSt24 you are checking validity and `textToCheck` is not used anywhere except for name so this seems like a method which requires some finishing touch while `handleTextInputWith` does same or pretty much similar task so we have duplicated code here.

I am wondering if there is something you can do to perform some sort of cleanup?

Reply:
This is also the reason for your failing tests, could you please cleanup the two methods

Reply:
@petruSt24 Pushed a helping commit here for cleanup - https://github.com/mozilla-mobile/firefox-ios/pull/15442/commits/b40006f345d09fec11b24eb801ff3ab03be2cd41

Try to write tests for these as they should be fairly straight forward to test

Reply:
yes I added that parameter as a helper with the unit tests but it didn't help as the text property can't be changed that way. I did more research about testing swiftUI views and apparently it's not as easy to test as UIKit so I updated the tests to check the viewModel.